app/models/course/video/tab.rb

```diff
@@ -24,6 +24,10 @@ def other_tabs_remaining?
     course.video_tabs.count > 1
   end
 
+  def initialize_duplicate(duplicator, other)
+    self.course = duplicator.options[:target_course]
```

Comment:
Could you add logic to check if any of the original tab's videos have been copied and associate them here?

This is particularly for object duplication. The idea is that it should not matter whether the video or its tab are duplicated first: as long as both are in the duplication list, their children should be associated. Specs for this can be added in `object_duplication_service_spec.rb`.

Reply:
Alright, added the association and tests.